### PR TITLE
Feat dotnet mutlipart

### PR DESCRIPTION
**Pull request by byawitz**
## What does this PR do?

(Provide a description of what this PR does.)

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work.)

## Related PRs and Issues

(If this PR is related to any other PR or resolves any issue or related to any issue link all related PR and issues here.)

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

(Write your answer here.)

### CODE REVIEW

File: src/SDK/Language/DotNet.php [hunk 1]
```diff
@@ -160,6 +160,10 @@ public function getPropertyOverrides(): array
      */
     public function getTypeName(array $parameter, array $spec = []): string
     {
+        if (strpos(($parameter['description'] ?? ''), 'This will return empty unless execution') !== false) {
```
Comment (Meldiron) [inline src/SDK/Language/DotNet.php:163]: Can we couple it with action name and param name instead?

**Review by Meldiron — APPROVED** [review]

File: templates/dotnet/Package/Models/Model.cs.twig [hunk 2]
```diff
@@ -40,7 +40,7 @@ namespace {{ spec.title | caseUcfirst }}.Models
...
... property.type == 'array' %}((JArray)map["{{ property.name }}"]).ToObject<{{ property | typeName }}>(){% else %}{% if property.type == "integer" or property.type == "number" %}{% if not property.required %}map["{{ property.name }}"] == null ? null : {% endif %}Convert.To{% if property.type == "integer" %}Int64{% else %}Double{% endif %}(map["{{ property.name }}"]){% else %}{% if property.type == "boolean" %}({{ property | typeName }}{% if not property.required %}?{% endif %})map["{{ property.name }}"]{% else %}map{% if not property.required %}.TryGetValue("{{ property.name }}", out var {{ property.name | caseCamel | escapeKeyword | removeDollarSign }}) ?  {{ property.name | caseCamel | escapeKeyword | removeDollarSign }}{% if property.name == "responseBody" or property.type | caseLower == "file" %} as Payload{% else %}?.ToString(){% endif %} : null{% else %}["{{ property.name }}"]{% if not property.required %}?{% endif %}{% if property.name == "responseBody" or property.type | caseLower == "payload" %} as Payload{% else %}.ToString(){% endif %}{% endif %}{% endif %}{% endif %}{% endif %}{% endif %}{% if not loop.last or (loop.last and definition.additionalProperties) %},{% endif %}
```
Comment (loks0n) [inline templates/dotnet/Package/Models/Model.cs.twig:43]:
```suggestion
{{ property.name | caseCamel | escapeKeyword | removeDollarSign }}:
{%~ if property.sub_schema %}
    {%~ if property.type == 'array' %}
        ((JArray)map["{{ property.name }}"])
            .ToObject<List<Dictionary<string, object>>>()
            .Select(it => {{property.sub_schema | caseUcfirst | overrideIdentifier}}.From(map: it))
            .ToList()
    {%~ else %}
        {{property.sub_schema | caseUcfirst | overrideIdentifier}}.From(
            map: ((JObject)map["{{ property.name }}"])
                .ToObject<Dictionary<string, object>>()!
        )
    {%~ endif %}
{%~ else %}
    {%~ if property.type == 'array' %}
        ((JArray)map["{{ property.name }}"])
            .ToObject<{{ property | typeName }}>()
    {%~ else %}
        {%~ if property.type == "integer" or property.type == "number" %}
            {%~ if not property.required -%}
                map["{{ property.name }}"] == null
                    ? null
                    : 
            {%~ endif -%}
            Convert.To{% if property.type == "integer" %}Int64{% else %}Double{% endif %}(map["{{ property.name }}"])
        {%~ else %}
            {%~ if property.type == "boolean" %}
                ({{ property | typeName }}{% if not property.required %}?{% endif %})map["{{ property.name }}"]
            {%~ else %}
                map{% if not property.required %}.TryGetValue("{{ property.name }}", out var {{ property.name | caseCamel | escapeKeyword | removeDollarSign }})
                    ? {{ property.name | caseCamel | escapeKeyword | removeDollarSign }}{% if property.name == "responseBody" or property.type | caseLower == "file" %} as Payload{% else %}?.ToString(){% endif %}
                    : null
                {%~ else %}
                    ["{{ property.name }}"]{% if not property.required %}?{% endif %}{% if property.name == "responseBody" or property.type | caseLower == "payload" %} as Payload{% else %}.ToString(){% endif %}
                {%~ endif %}
            {%~ endif %}
        {%~ endif %}
    {%~ endif %}
{%~ endif %}{% if not loop.last or (loop.last and definition.additionalProperties) %},{% endif -%}
```

File: tests/resources/spec.json [hunk 3]
```diff
@@ -1574,11 +1577,8 @@
         ],
         "description": "",
         "responses": {
-          "200": {
-            "description": "Multipart",
-            "schema": {
-              "$ref": "#\/definitions\/multipart"
-            }
+          "301": {
```
Comment (loks0n) [inline tests/resources/spec.json]: Can we revert this?

Comment (Meldiron) [reply tests/resources/spec.json]: Reverting

File: templates/dotnet/Package/Models/Model.cs.twig (hunk 2 above)
Comment (Meldiron) [reply templates/dotnet/Package/Models/Model.cs.twig:43]: I like this but I think it should be done as part of code quality project panned later this week. We have issue for it.

**Pull request merged**